src/adapter/src/catalog/migrate.rs

```diff
@@ -226,3 +226,44 @@ fn ast_rewrite_create_source_loadgen_options_0_92_0(
 
     Ok(())
 }
+
+// Durable migrations
+
+/// Migrations that run only on the durable catalog before any data is loaded into memory.
+pub(crate) fn durable_migrate(tx: &mut Transaction) -> Result<(), anyhow::Error> {
+    catalog_fix_system_cluster_replica_ids_v_0_95_0(tx)?;
+    Ok(())
+}
+
+// Add new migrations below their appropriate heading, and precede them with a
+// short summary of the migration's purpose and optional additional commentary
+// about safety or approach.
+//
+// The convention is to name the migration function using snake case:
+// > <category>_<description>_<version>
+//
+// Please include the adapter team on any code reviews that add or edit
+// migrations.
+
+fn catalog_fix_system_cluster_replica_ids_v_0_95_0(
+    tx: &mut Transaction,
+) -> Result<(), anyhow::Error> {
+    let updated_replicas: Vec<_> = tx
+        .get_cluster_replicas()
+        .filter(|replica| replica.cluster_id.is_system() && replica.replica_id.is_user())
+        .map(|replica| (replica.replica_id, replica))
+        .collect();
+    for (replica_id, mut updated_replica) in updated_replicas {
+        let sys_id = tx.allocate_system_replica_id()?;
+        updated_replica.replica_id = sys_id;
+        tx.remove_cluster_replica(replica_id)?;
```

Comment:
Do we need to care about audit log events here? In the past we have made drop and create events for migrations like these. But maybe we don't need to if the initial replica doesn't get an event because it's on a system cluster or something? I think: if there is an existing audit log event for this replica then we need audit logs for this migration.

Reply:
Good point, I'll add something.

Reply:
Hmm, it seems like we've lost access to the timestamp oracle in migrations which is needed to produce a linearizable timestamp which is needed to add audit log entries. I'll have to figure out what to do about that.